src/app/data/components/data-browser/data-browser.component.html

```diff
@@ -1288,4 +1004,241 @@ <h1 class="text-lg text-gray-900 mb-2">Info</h1>
+        Save or update a filter for the data browser.</div>
+    <p class="pt-4">You have two choices.
+        <label class="block font-bold mt-2">Dynamic:</label>
+        Stores the current filter critera and is reapplied whenever the slice is
+        selected.<br><br>
+        <label class="block font-bold mt-2 ">Static:</label>
+        Used for visualization filters and scope of
+        work for annotators & domain experts.
+        Stores the current records provided by your filter.
+        These are limited to 10,000 records and can be refreshed if nessecary.
+    </p>
+    <div class="mt-2">
+        <div class="flex">
+            <div class="inline-flex">
+                <div class="form-control">
+                    <label class="cursor-pointer label">
+                        <span class="label-text mr-1">Dynamic</span>
+                        <input #inputDynamic type="radio" style="--focus-shadow:none;accent-color:rgb(29,78,216)"
+                            (click)="setStatic(inputDynamic,inputStatic,false)"
+                            class="h-4 w-4 border-gray-200 cursor-pointer" checked [value]="true">
+                    </label>
+                </div>
+                <div class="form-control">
+                    <label class="cursor-pointer label">
+                        <span class="label-text mr-1">Static</span>
+                        <input #inputStatic type="radio" style="--focus-shadow:none;accent-color:rgb(180,83,9)"
+                            (click)="setStatic(inputDynamic,inputStatic,true)"
+                            class="h-4 w-4 border-gray-200 cursor-pointer" [value]="false">
+                    </label>
+                </div>
+            </div>
+            <div data-tip="Too many records for a static slice" class="relative tooltip mr-1"
+                style="white-space: break-spaces;">
+                <div class="flex justify-center items-center w-full h-full"
+                    [ngClass]="{hidden:!dataBrowserModals.filter.displayStaticNotAllowedWarning}">
+                    <svg xmlns="http://www.w3.org/2000/svg" class="h-6 w-6" fill="none" viewBox="0 0 24 24"
+                        stroke="currentColor" stroke-width="2">
+                        <path fill="#E8CD20" stroke-linecap="round" stroke-linejoin="round"
+                            d="M12 9v2m0 4h.01m-6.938 4h13.856c1.54 0 2.502-1.667 1.732-3L13.732 4c-.77-1.333-2.694-1.333-3.464 0L3.34 16c-.77 1.333.192 3 1.732 3z" />
+                    </svg>
+                </div>
+            </div>
+        </div>
+        <div class="flex w-full justify-between mt-2">
+            <input #inputSliceName type="text" class="w-1/2 input input-bordered input-sm" placeholder="Enter name"
+                name="dataSliceName" [value]="dataBrowserModals.filter.name"
+                (input)="setFilterName(inputSliceName.value)" (keyup)="checkNameExists(inputSliceName.value)"
```

Comment:
not quite sure why these are in different methods both should work in keyup.
Also setting the name should only be necessary right before accepting, right?

Reply:
Yes, that is true, both should be in keyup. 
Also yes for the second thing, I will move the logic from setFilterName to both methods for update and create slice.